lib/iris/tests/unit/analysis/area_weighted/test_AreaWeightedRegridder.py

```diff
@@ -193,37 +199,39 @@ def test_multiple_src_on_same_grid(self):
         self.assertEqual(result1, reference1)
         self.assertEqual(result2, reference2)
 
-    def test_mismatched_data_dims(self):
-        coord_names = ["latitude", "longitude"]
-        x = np.linspace(20, 32, 4)
-        y = np.linspace(10, 22, 4)
-        src1 = self.cube(x, y)
-
-        data = np.arange(len(y) * len(x)).reshape(len(x), len(y))
-        src2 = Cube(data)
-        lat = DimCoord(y, "latitude", units="degrees")
-        lon = DimCoord(x, "longitude", units="degrees")
-        # Add dim coords in opposite order to self.cube.
-        src2.add_dim_coord(lat, 1)
-        src2.add_dim_coord(lon, 0)
-        for name in coord_names:
-            # Ensure contiguous bounds exists.
-            src1.coord(name).guess_bounds()
-            src2.coord(name).guess_bounds()
-
-        target = self.cube(np.linspace(20, 32, 2), np.linspace(10, 22, 2))
-        for name in coord_names:
-            # Ensure the bounds of the target cover the same range as the
-            # source.
-            target.coord(name).bounds = np.column_stack(
-                (
-                    src1.coord(name).bounds[[0, 1], [0, 1]],
-                    src1.coord(name).bounds[[2, 3], [0, 1]],
-                )
-            )
-
-        regridder = AreaWeightedRegridder(src1, target)
-        self.assertRaises(ValueError, regridder, src2)
+    def test_src_data_different_dims(self):
+        src, target = self.grids()
+        regridder = AreaWeightedRegridder(src, target)
+        result = regridder(src)
+        expected_mean, expected_std = 4.772097735195653, 2.211698479817678
+        self.assertArrayShapeStats(result, (9, 8), expected_mean, expected_std)
+        # New source cube with additional "levels" dimension
+        # Each level has identical x-y data so the mean and std stats remain
+        # identical when x, y and z dims are reordered
+        levels = DimCoord(np.arange(5), "model_level_number")
+        lat = src.coord("latitude")
+        lon = src.coord("longitude")
+        data = np.repeat(src.data[np.newaxis, ...], 5, axis=0)
+        src = Cube(data)
+        src.add_dim_coord(levels, 0)
+        src.add_dim_coord(lat, 1)
+        src.add_dim_coord(lon, 2)
+        result = regridder(src)
+        self.assertArrayShapeStats(
+            result, (5, 9, 8), expected_mean, expected_std
+        )
+        # Check data with dims in different order
+        # Reshape src so that the coords are ordered [x, z, y],
```

Comment:
There is no longer a case tested in which x is assigned to a higher dimension than y e.g.`[y, x, z]`. Given that these two cases are treated differently in the code (specifically, the code in `_regrid_area_weighted_array`), both of these cases should probably be covered in the tests.